openapi.yml

```diff
@@ -0,0 +1,447 @@
+        - "$ref": "#/components/parameters/api-entity-header"
+        - "$ref": "#/components/parameters/api-secret-header"
+        - "$ref": "#/components/parameters/api-user-header"
+        - in: query
+          name: sequential
+          schema:
+            type: boolean
+          description: Indicates whether IDs should be sequential
+        - in: query
+          name: amount
+          schema:
+            type: integer
+          description: Amount of IDs desired
+      responses:
+        200:
+          description: A list of the newly reserved CVE-IDs the account requested.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/cve-id'
+        400:
+          description: Bad Request
+        401:
+          description: Not Authenticated
+        403:
+          description: Forbidden
+        404:
+          description: Not Found
+        500:
+          description: Internal Server Error
+  /cve-id/{cve-id}:
+    parameters:
+      - "$ref": "#/components/parameters/cve-id-path"
+    get:
+      summary: Retrieves a CVE ID by ID, can be a CVE the entity doesn't own if in a PUBLIC or REJECT state.
+      operationId: getCVEIdById
+      parameters:
+        - "$ref": "#/components/parameters/api-entity-header"
+        - "$ref": "#/components/parameters/api-secret-header"
+        - "$ref": "#/components/parameters/api-user-header"
+      responses:
+        200:
+          description: >
+            The requested CVE ID is returned.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/cve-id'
+        400:
+          description: Bad Request
+        401:
+          description: Not Authenticated
+        403:
+          description: Forbidden
+        404:
+          description: Not Found
+        500:
+          description: Internal Server Error
+    # Posts only accessible by Secretariat
+    post:
+      summary: Endpoint for Admin purposes to change state and ownership of the CVE ID.
+      operationId: submitCVEIdById
+      parameters:
+        - "$ref": "#/components/parameters/api-entity-header"
+        - "$ref": "#/components/parameters/api-secret-header"
+        - "$ref": "#/components/parameters/api-user-header"
+        - in: query
+          name: state
+          schema:
+            type: string
+          description: "Change state of CVE ID to either PUBLIC or REJECT"
+        - in: query
+          name: cna
+          schema:
+            type: string
+          description: "Change owning CNA of CVE ID to new CNA (shortname)"
+      responses:
+        200:
+          description: >
+            The updated CVE ID is returned.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/cve-id'
+        400:
+          description: Bad Request
+        401:
```

Comment:
Is this returned from the endpoint or from the middleware?

Reply:
Same as my other response